src/components/OptionsBar/Options/FilterModal/Filters/RelativeFilters.jsx

```diff
@@ -4,43 +4,43 @@ import { useRef } from 'react';
 const relativeFilters = (filterIds, tabIndex) => {
   const filters = useSelector((state) => state.filters);
 
-  const getFilterMatrix = () => {
-    const filterMatrix = new Array(filterIds.length);
-
-    filterIds.forEach((id, index) => {
-      filterMatrix[index] = filters[id];
-    });
-
-    return filterMatrix;
-  };
+  const getFilterMatrix = () => (filters ? filterIds.map((id) => filters?.[id]) : []);
 
   const queuedFilters = useRef(getFilterMatrix());
 
   const getFilters = () => queuedFilters.current;
 
   const getRelativeFilters = () => getFilters()[tabIndex];
 
-  const pushRelativeFilters = (filter) => getRelativeFilters().push(filter);
+  const setRelativeFilters = (value) => {
+    queuedFilters.current[tabIndex] = value;
+
+    return getRelativeFilters();
+  };
 
-  const spliceRelativeFiltersAt = (index) => getRelativeFilters().splice(index, 1);
+  const pushRelativeFilters = (filter) => {
+    setRelativeFilters([ ...getRelativeFilters(), filter ]);
 
-  const rebuildFilters = () => {
-    getFilters().forEach((queuedFilter, index) => {
-      getFilters()[index] = [ ...getFilters()[index] ];
-    });
+    return getRelativeFilters();
+  };
+
+  const spliceRelativeFiltersAt = (index) => {
+    setRelativeFilters([ ...getRelativeFilters().filter((filter, i) => i !== index) ]);
```

Comment:
You dont need to spread the array, because filter already returns you a new copy. Also for this long statement, it would be better to assign it first to a const

```suggestion
    const matchingFilters = getRelativeFilters().filter((filter, i) => i !== index);
    setRelativeFilters(matchingFilters);
```